simulation/utils/generate_map.py

```diff
@@ -143,54 +164,51 @@ class Difficulty(Enum):
     REALLY_HARD = 4
     WTF = 5
 
-# ************
-def intersect(A, B, C, D):
-    """Helper of is_crossing function
-    Returns True if line segments AB and CD intersect"""
-    ccw = lambda A, B, C: (C.y - A.y) * (B.x - A.x) > (B.y - A.y) * (C.x - A.x)
-    return ccw(A, C, D) != ccw(B, C, D) and ccw(A, B, C) != ccw(A, B, D)
-
-# ************
-def next_point(prev_point, step, alpha, y_res):
+    
+def generate_next_point(prev_point, step, alpha, y_resolution):
     """Function returning next random point for the generalized map"""
     y_range = 2 * step * math.tan(alpha / 2)
-    ymin = prev_point.y - y_range / 2
-    ymin = ymin if ymin >= 0 else 0
-    ymax = prev_point.y + y_range / 2
-    ymax = ymax if ymax < y_res - 20 else y_res - 20
+
+    y_min = prev_point.y - y_range / 2
+    y_min = y_min if y_min >= 0 else 0
+    y_max = prev_point.y + y_range / 2
+    y_max = y_max if y_max < y_resolution - Y_CEILING_LIMIT else y_resolution - Y_CEILING_LIMIT
+
     p = random.random()
     if p >= 0.3:
```

Comment:
To 0.3 z czegoś wynika, czy to po prostu jakiś założony parametr?

Reply:
Jak robiłam te funkcje to sprawdzalam różne parametry, i z tym najlepiej wychodziło, ale masz rację, zrobię z tego też globalną zmienną

Reply:
🥇